static/js/dialog_widget.js

```diff
@@ -157,7 +158,24 @@ export function launch(conf) {
             }
         },
         on_hide: conf?.on_hide,
-        on_shown: conf?.on_shown,
+        on_shown: () => {
+            const modal_scroll_element = document.querySelector(
+                "#dialog_widget_modal .simplebar-content-wrapper",
+            );
+            const $modal_footer = $dialog.find(".modal__footer");
+            if (modal_scroll_element.scrollHeight > modal_scroll_element.clientHeight) {
+                const using_dark_theme = settings_data.using_dark_theme();
+                if (using_dark_theme) {
+                    $modal_footer.css("border-top", "1px solid hsla(0, 0%, 100%, 0.2)");
+                } else {
+                    $modal_footer.css("border-top", "1px solid hsl(0, 0%, 87%)");
+                }
+            }
```

Comment:
There's a few things we need to change here:
* We should set a `scrollable_modal` CSS class, it would be better to have `border-top` styling for the two themes in CSS files, rather than doing inline `.css` here. Among other things, doing so will live-update properly, and also just have all the CSS remain in CSS files.
* Some modals, like Read Receipts, can change in height after being initially loaded (when a network request returns)
* All modals can have the browser window be resized while they're open; and this logic likely won't handle that.

A pure CSS solution would be a lot better; and I think it might be possible; there should be something like a `:visible` selector we can do on a simplebar generated container for whether a scrollbar is visible or not.

Reply:
Not sure what a pure CSS solution would be. I went with using a `MutationObserver` on the vertical scrollbar. Updated the PR.